### ISSUE TITLE
gpu not used by mesa driver

### ISSUE THREAD

**Issue by LuanVSO** [sileshn/ManjaroWSL2#1]
### Requirements

- [X] Yes
- [ ] No

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Version

10.0.22621.819

### Kernel Version

5.15.74.2

### Repro Steps

WSL Version: 1.0.0.0
kernel Version: 5.15.74.2
WSLg Version: 1.0.47
MSRDC Version: 1.2.3575
Direct3D Version: 1.606.4
DXCore Version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows Version: 10.0.22621.819

---

```bash
pacman -S mesa mesa-utils
glxgears
```
check on windows taskmanager to see if the gpu is being used.

### Expected Behavior

the gpu is used to render glxgears

### Actual Behavior

glxgears is rendered using cpu

---
### additional info:
```
❯ glxinfo | grep -i opengl
OpenGL vendor string: Mesa/X.org
OpenGL renderer string: llvmpipe (LLVM 14.0.6, 256 bits)
OpenGL core profile version string: 4.5 (Core Profile) Mesa 22.2.3
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 4.5 (Compatibility Profile) Mesa 22.2.3
OpenGL shading language version string: 4.50
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.2 Mesa 22.2.3
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```
```
❯ lspci
94ed:00:00.0 3D controller: Microsoft Corporation Basic Render Driver

```

**Comment by sileshn** [comment]
I am not able to reproduce this.

![Screenshot 2022-11-23 035423](https://user-images.githubusercontent.com/977360/203433699-5165be97-7d46-484a-aa6e-dbbbed0a16bd.png)

**Comment by LuanVSO** [comment]
forgot to mention that i am on intel integrated graphics, the intel gpu info only shows up on ubuntu:
![image](https://user-images.githubusercontent.com/43626415/203441716-9442c48f-2792-47b4-a6cc-40453dc5c9a6.png)

**Comment by sileshn** [comment]
Irrespective of the GPU being amd/intel/nvidia, it should work as long as you have the correct drivers installed

**Comment by LuanVSO** [comment]
i re-installed the distro and only installed the mesa packages, it still doesn't work, do you have any pointers/tips on how to debug this?
![image](https://user-images.githubusercontent.com/43626415/204051957-0fad331c-f2c7-43c7-abac-41bee8ebed65.png)

**Comment by sileshn** [comment]
Link me to it

**Comment by Vinfall** [comment]
Sorry, I just recall it and can't remember the source... But I can point you some links that may help:

1. [WDDM GPU Paravirtualization support for WSL2](https://github.com/intel/compute-runtime/blob/master/WSL.md#wddm-gpu-paravirtualization-support-for-wsl2): general guide
2. [Configure the host to bind vfio-pci to Intel Iris Xe](https://dgpu-docs.intel.com/devices/iris-xe-max-graphics/index.html#intel-iris-xe-max-graphics-with-linux): more detailed but maybe exclusive to Linux desktop or Iris Xe, no idea whether it applies to WSL or UHD. It involves compiling the kernel if earlier than Ubuntu 22.10

Basically it's either vGPU driver not installed/working or kernel w/o the driver adapter enabled.
